.github/workflows/ci.yml

```
@@ -153,3 +137,13 @@ jobs:
         run: python ./prereqs.py --install
       - name: Run integration tests
         run: python ./build.py --no-check --no-test --wasm --npm --vscode --integration-tests
+
+  runBenchmark:
+    name: run benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: boa-dev/criterion-compare-action@v3
+        with:
+          branchName: ${{ github.base_ref }}
+          cwd: "compiler/qsc"
```

I think you can add a condition here to make it play nice with merge queues:


```suggestion
          cwd: "compiler/qsc"
        if: ${{ github.base_ref != null }}
```